module/entities/helper.js

```diff
@@ -595,4 +596,110 @@ module.exports = class EntitiesHelper {
 
 }
 
+    /**
+   * Get users by entityId and role
+   * @method
+   * @name getUsersByEntityAndRole
+   * @param {Object} requestedData - requested data.
+   * @param {String} req.body.entityId - entity id.
+   * @param {String} req.body.role - role code.
+   * @returns {Array}  - List of userIds and entityIds
+   */
+
+  static getUsersByEntityAndRole( entityId= "", role= "" ) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if (entityId == "") {
+                throw new Error(constants.apiResponses.ENTITY_ID_REQUIRED);
+            }
+
+            if (role == "") {
+                throw new Error(constants.apiResponses.ROLE_REQUIRED)
+            }
+            
+            let entityDocument = await this.entityDocuments
+            (
+                {
+                    _id: entityId
+                },
+                [
+                    "entityType" 
+                ]
+            )
+            
+            if (!entityDocument.length) {
+                throw new Error(constants.apiResponses.INVALID_ENTITY_ID);
+            }
+
+            let userRoles = await userRolesHelper.roleDocuments({
+                code : role
+            },[
+                "entityTypes"
+            ]);
+            
+            if(!userRoles.length) {
+                throw new Error(constants.apiResponses.INVALID_ROLE)
+            }
+
+            let queryObject = {
+                query: {
+                    bool: {
+                        must: [
+                            { match: { "data.entityType": userRoles[0].entityTypes[0].entityType } }
+                        ]
+                    }
+                },
+                _source: [`data.roles.${role}`]
+            }
+            
+            let entityMatch = {match : {}};
+            entityMatch.match[`data.telemetry_entities.${entityDocument[0].entityType}_id`] = entityId;
+
+            queryObject.query.bool.must.push(entityMatch);
+           
+            let entityDocuments = await elasticSearch.searchDocumentFromIndex
+            (
+                process.env.ELASTICSEARCH_ENTITIES_INDEX,
+                "_doc",
+                queryObject,
+                null,
```

Comment:
@deepakr123 - Do we need to pass null ? Just don't pass those parameters

Reply:
if we don't pass, default size 10 is being set which is used by some other API. SO i should pass null.